vsintegration/src/FSharp.Editor/LanguageService/FSharpProjectOptionsManager.fs

```diff
@@ -132,49 +152,41 @@ type private FSharpProjectOptionsReactor (workspace: VisualStudioWorkspaceImpl,
             | true, (fileStamp2, parsingOptions, projectOptions) ->
                 if fileStamp <> fileStamp2 then
                     singleFileCache.Remove(document.Id) |> ignore
-                    return! tryComputeOptionsByFile document
+                    return! tryComputeOptionsByFile document cancellationToken
                 else
                     return Some(parsingOptions, projectOptions)
         }
     
-    let rec tryComputeOptions (project: Project) =
+    let rec tryComputeOptions (project: Project) (cancellationToken: CancellationToken) = async {
         let projectId = project.Id
-        let projectStamp = project.Version
         match cache.TryGetValue(projectId) with
         | false, _ ->
 
             // Because this code can be kicked off before the hack, HandleCommandLineChanges, occurs,
             //     the command line options will not be available and we should bail if one of the project references does not give us anything.
             let mutable canBail = false
+            
+            let referencedProjects = ResizeArray()
 
-            let referencedProjects =
-                if settings.LanguageServicePerformance.EnableInMemoryCrossProjectReferences then
+            if settings.LanguageServicePerformance.EnableInMemoryCrossProjectReferences then
+                let computeOptionsAsyncArray =
                     project.ProjectReferences
                     |> Seq.choose (fun projectReference -> 
                         let referenceProject = project.Solution.GetProject(projectReference.ProjectId)
                         if referenceProject.Language = FSharpConstants.FSharpLanguageName then
-                            Some(referenceProject)
+                            Some((async { return! tryComputeOptions referenceProject cancellationToken }, referenceProject))
```

Comment:
Feels kinda funky having the function embedded like that, even if it's just one line.

Reply:
It's for delayed computation so I can easily iterate in a `for.. in...`

Reply:
Yup, this whole thing was funky. About to update it.